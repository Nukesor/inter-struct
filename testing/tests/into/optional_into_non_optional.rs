use inter_struct::InterStruct;

/// This should crash, since the generated initializer for IntoStruct will be incomplete.
#[derive(InterStruct)]
#[into("crate::IntoStruct")]
pub struct FromStruct {
    pub normal: Option<String>,
    pub optional: Option<Option<String>>,
}

pub struct IntoStruct {
    pub normal: String,
    pub optional: Option<String>,
}

fn main() {}
