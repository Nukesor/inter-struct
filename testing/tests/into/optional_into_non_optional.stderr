error: Inter-struct cannot 'into' an optional into a non-optional value.
 --> tests/into/optional_into_non_optional.rs:7:24
  |
7 |     pub normal: Option<String>,
  |                        ^^^^^^

error: Inter-struct cannot 'into' an optional into a non-optional value.
 --> tests/into/optional_into_non_optional.rs:8:26
  |
8 |     pub optional: Option<Option<String>>,
  |                          ^^^^^^

error[E0063]: missing fields `normal` and `optional` in initializer of `IntoStruct`
 --> tests/into/optional_into_non_optional.rs:5:8
  |
5 | #[into("crate::IntoStruct")]
  |        ^^^^^^^^^^^^^^^^^^^ missing `normal` and `optional`
